### DISCUSSION TITLE
Next Auth Redirecting me to /undefined and gives me strange error

### DISCUSSION THREAD

**Discussion by nicolaeser** [nextauthjs/next-auth#9954] [Help]
`auth.ts`
![grafik](https://github.com/nextauthjs/next-auth/assets/67909777/2554ab32-34a0-4a7d-87d8-0199b97fdf6d)

`/api/auth/[...nextauth]/route.js`
![grafik](https://github.com/nextauthjs/next-auth/assets/67909777/14a1a4e3-f663-4fc8-9812-ffa528a156ea)

# Error
![grafik](https://github.com/nextauthjs/next-auth/assets/67909777/9351480c-e9b2-4869-b835-5465da3b2dbd)

# My Auth Button Component
![grafik](https://github.com/nextauthjs/next-auth/assets/67909777/b3c19ae7-7822-493c-8b62-32095e73f703)

### That really worked for weeks and i thought all errors would be gone... 
![grafik](https://github.com/nextauthjs/next-auth/assets/67909777/70d68c32-1e6d-4581-943a-a26fff226aa2)

**Comment by pippinmole** [comment]
Hi,

Authjs changed the environment variables slightly. You need to do the following:
1. Remove `NEXTAUTH_URL`. This is no longer used, [see this comment by the author of the library](https://github.com/nextauthjs/next-auth/pull/9955#issue-2124101197).
2. Rename `NEXTAUTH_SECRET` to `AUTH_SECRET`, [see this link](https://github.com/nextauthjs/next-auth/pull/9955#issue-2124101197).

After applying these two modifications, it should be fixed.
